src/devices/Bmxx80/Bmx280Base.cs

```diff
@@ -0,0 +1,291 @@
+            byte current = Read8BitsFromRegister((byte)Bmx280Register.CONFIG);
+            var mode = (byte)((current & 0b_0001_1100) >> 2);
+
+            return mode switch
+            {
+                0b000 => FilteringMode.Off,
+                0b001 => FilteringMode.X2,
+                0b010 => FilteringMode.X4,
+                0b011 => FilteringMode.X8,
+                _ => FilteringMode.X16
+            };
+        }
+
+        /// <summary>
+        /// Read the temperature.
+        /// </summary>
+        /// <returns>Calculated temperature.</returns>
+        public async Task<Temperature> ReadTemperatureAsync()
+        {
+            if (ReadPowerMode() == Bmx280PowerMode.Forced)
+            {
+                await Task.Delay(GetMeasurementTimeForForcedMode(ReadTemperatureSampling()));
+            }
+
+            //Read the MSB, LSB and bits 7:4 (XLSB) of the temperature from the BMP280 registers
+            byte msb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_MSB);
+            byte lsb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_LSB);
+            byte xlsb = Read8BitsFromRegister((byte)Bmx280Register.TEMPDATA_XLSB); // bits 7:4
+
+            //Combine the values into a 32-bit integer
+            int t = (msb << 12) + (lsb << 4) + (xlsb >> 4);
+
+            return CompensateTemperature(t);
+        }
+
+        /// <summary>
+        /// Read the <see cref="Bmx280PowerMode"/> state.
+        /// </summary>
+        /// <returns>The current <see cref="Bmx280PowerMode"/>.</returns>
+        /// <exception cref="NotImplementedException">Thrown when the power mode does not match a defined mode in <see cref="Bmx280PowerMode"/>.</exception>
+        public Bmx280PowerMode ReadPowerMode()
+        {
+            byte read = Read8BitsFromRegister(_controlRegister);
+
+            // Get only the power mode bits.
+            var powerMode = (byte)(read & 0b_0000_0011);
+
+            if (Enum.IsDefined(typeof(Bmx280PowerMode), powerMode) == false)
+            {
+                throw new IOException("Read unexpected power mode");
+            }
+
+            return powerMode switch
+            {
+                0b00 => Bmx280PowerMode.Sleep,
+                0b10 => Bmx280PowerMode.Forced,
+                0b11 => Bmx280PowerMode.Normal,
+                _ => throw new NotImplementedException($"Read power mode not defined by specification.")
+            };
+        }
+
+        /// <summary>
+        /// Reads the pressure from the sensor.
+        /// </summary>
+        /// <returns>Atmospheric pressure in Pa.</returns>
+        public async Task<double> ReadPressureAsync()
+        {
+            if (ReadPowerMode() == Bmx280PowerMode.Forced)
+            {
+                await Task.Delay(GetMeasurementTimeForForcedMode(ReadPressureSampling()));
+            }
+
+            //Read the temperature first to load the t_fine value for compensation
+            if (TemperatureFine == int.MinValue)
```

Comment:
I might be missing it but where is TemperatureFine being initialized to int.MinValue? I don't see that happening in the constructor so technically I believe this guy will be defaulted to 0 right? Should we initialize it to int.MinValue in the constructor of this base class?

Reply:
Yup, good spot! Shall initialise in the constructor of this base class.